src/decision/location_selection.jl

```diff
@@ -124,7 +125,14 @@ function rank_locations(
         depth_priority = findall(depth_criteria)
 
         considered_sites = target_site_ids[findall(in(depth_priority), target_site_ids)]
-        mcda_vars_temp = DMCDA_vars(domain, scen, considered_sites,  sum_cover[scen_idx, :], area_to_seed,
+        mcda_vars_temp = DMCDA_vars(
+            domain,
+            scen,
+            considered_sites,
+            relative_leftover_space(
```

Comment:
I personally am very against calling one function as argument to another the way is happening here because of readability. I'm not sure how much performance you lose by assigning this to a variable instead since Julia will have to allocate memory to do this calculation anyways (I THINK). 

@ConnectedSystems thoughts?

Reply:
We could change MCDAvars but the variable will have to be stripped at somepoint because it is put into the decision matrix

Reply:
I'd then ask: why doesnt the decision matrix use named dimensions? Relying on roe/column indices is a potential source of bugs

Reply:
This was one of the changes I was pushing for in the PR to make MCDA generic, but that got broken into smaller PRs and this aspect still hasn't been done. Agree it's a priority, but probably not for this PR.

Actually, I realise now it's because `relative_leftover_space` expects a vector or Matrix that I have to strip the NmaedDimsArray here

Reply:
Using `Matrix()` errors because it's a row, so I could use Array or collect instead. I've gone with collect, but let me know if Array is better.

Reply:
If it is a row then it's Vector is more explicit than Array, no?

Reply:
As in `vec()`? - that is still a NamedDimsArray though with a single named dimension

Reply:
A few comments:

1. Maybe `relative_leftover_space` should expect/return an `AbstractArray`, this way it will work with Vectors, Matrices, NamedDimsArrays, etc. Also, this way, in the future, when you make the decision matrix use named dimensions, this function won't have to be changed.
2. Suppose you have `nda = NamedDimsArray(rand(2,2), x=[1,2], y=[3,4])`, you can do `nda.data.data` to retrieve the raw matrix, and you can also do `nda[1,:].data.data` to get your vector. I think this could be a temporary solution while we wait for the decision matrix and `DMCDA_vars` changes, so maybe creating an issue to come back here and change that later would be interesting?

Reply:
Using `nda[1,:].data.data` isn't working for me. Would it be : `sum_cover[scen_idx, :].data.data` ?

Reply:
Hum.. .yes... why does it work for me then..? Can you try `using NamedDims` and `using AxisKeys` just to confirm? 

Also, when you do `sum_cover[scen_idx, :].data` what happens?

Reply:
I'll try this

Reply:
Just `sum_cover[scen_idx, :].data` works :) I'll push the changes

Reply:
All that said, I still think this should be assigned to a separate variable for readability (:

Reply:
Thanks @Zapiano, realised I should have pre-allocated `leftover_space` outside the for loop anyway, so I've added that now. All tests passing, let me know if there's anything else but I think it's ready